core/common/src/main/java/alluxio/network/protocol/databuffer/NioDataBuffer.java

```diff
@@ -68,7 +68,14 @@ public void readBytes(OutputStream outputStream, int length) throws IOException
 
   @Override
   public void readBytes(ByteBuffer outputBuf) {
-    outputBuf.put(mBuffer);
+    if (mBuffer.remaining() <= outputBuf.remaining()) {
+      outputBuf.put(mBuffer);
+    } else {
+      int oldLimit = mBuffer.limit();
```

Comment:
This looks tricky. Could you comment why we have to do this? It looks like we change the limit of the output, write the data, then change the limit back? Is that safe to do for the output?

Reply:
in the case where `outputBuf` is smaller than the this databuffer,
it will fail when `outputBuf.put(mBuffer);`. 
So we need to tentatively make sure `outputBuf.put(mBuffer);` can proceed